crash

### Jerboa Version

0.0.77

### Android Version + Phone

Android 15, GrapheneOS, Pixel 6a

### Describe The Bug

Trying to switch to a different user account from the sidebar

### To Reproduce

1. Open jerboa
2. Switch to a different user

### In the case of a crash or when relevant include the logs

```
type: crash
osVersion: google/bluejay/bluejay:15/AP4A.250205.002/2025021100:user/release-keys
flags: dev options enabled
package: com.jerboa:77, targetSdk 34
process: com.jerboa
processUptime: 11342 + 56 ms
installer: dev.imranr.obtainium

coil.network.HttpException: Exception in completion handler InvokeOnCompletion@9f08cd3[job@4867010] for StandaloneCoroutine{Completed}@4867010
	at kotlinx.coroutines.JobSupport.completeStateFinalization(Unknown Source:137)
	at kotlinx.coroutines.JobSupport.finalizeFinishingState(Unknown Source:155)
	at kotlinx.coroutines.JobSupport$ChildCompletion.invoke(Unknown Source:41)
	at kotlinx.coroutines.JobSupport.completeStateFinalization(Unknown Source:44)
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(Unknown Source:53)
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(Unknown Source:4)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(Unknown Source:13)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:31)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:112)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(Unknown Source:140)
	at kotlin.io.ByteStreamsKt.runBlocking(Unknown Source:104)
	at androidx.appcompat.view.menu.CascadingMenuPopup$3$1.run(Unknown Source:33)
	at androidx.room.TransactionExecutor$$ExternalSyntheticLambda0.run(Unknown Source:110)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1117)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Completed}@4867010, Dispatchers.Main.immediate]
Caused by: java.lang.IllegalStateException: Method setCurrentState must be called on the main thread
	at androidx.lifecycle.LifecycleRegistry.enforceMainThreadIfNeeded(Unknown Source:42)
	at androidx.lifecycle.LifecycleRegistry.setCurrentState(Unknown Source:7)
	at androidx.navigation.NavBackStackEntry.updateState(Unknown Source:48)
	at androidx.navigation.NavBackStackEntry.setMaxLifecycle(Unknown Source:7)
	at androidx.navigation.NavController.popEntryFromBackStack(Unknown Source:107)
	at androidx.navigation.NavController$executePopOperations$1.invoke(Unknown Source:22)
	at androidx.navigation.NavController$NavControllerNavigatorState.pop(Unknown Source:38)
	at androidx.navigation.NavController$NavControllerNavigatorState.popWithTransition(Unknown Source:189)
	at androidx.navigation.compose.ComposeNavigator.popBackStack(Unknown Source:4)
	at androidx.navigation.NavController.executePopOperations(Unknown Source:61)
	at androidx.navigation.NavController.popBackStackInternal(Unknown Source:95)
	at androidx.navigation.NavController.navigate(SourceFile:47)
	at androidx.navigation.NavController.navigate$default(Unknown Source:84)
	at androidx.navigation.NavController.navigate(SourceFile:139)
	at com.jerboa.UtilsKt$$ExternalSyntheticLambda7.invoke(Unknown Source:81)
	at com.jerboa.ui.components.home.BottomNavScreenKt$$ExternalSyntheticLambda3.invoke(Unknown Source:176)
	at com.jerboa.ui.components.drawer.MainDrawerKt$$ExternalSyntheticLambda9.invoke(Unknown Source:28)
	at kotlinx.coroutines.InternalCompletionHandler$UserSupplied.invoke(Unknown Source:2)
	at kotlinx.coroutines.InvokeOnCompletion.invoke(Unknown Source:55)
	at kotlinx.coroutines.JobSupport.completeStateFinalization(Unknown Source:106)
	... 15 more
```

```
type: logcat
osVersion: google/bluejay/bluejay:15/AP4A.250205.002/2025021100:user/release-keys
flags: dev options enabled
package: com.jerboa:77, targetSdk 34
buffers: main,system,crash,events,kernel
level: verbose

--------- beginning of events
02-13 08:22:27.456 10644 10644 I wm_on_create_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=performCreate,time=12ms]
02-13 08:22:27.456 10644 10644 I wm_on_start_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=handleStartActivity,time=1ms]
02-13 08:22:27.458 10644 10644 I wm_on_resume_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=RESUME_ACTIVITY,time=2ms]
02-13 08:22:27.465 10644 10644 I wm_on_top_resumed_gained_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=topStateChangedWhenResumed]
02-13 08:22:27.566 10644 10644 I viewroot_draw_event: [window=VRI[MainActivity],event=reportDrawFinished seqId=0]
--------- beginning of main
02-13 08:22:38.591 10644 10644 D setLemmyInstance: Setting lemmy instance to https://feddit.org
02-13 08:22:38.594 10644 10644 D PostsViewModel: Fetching posts
02-13 08:22:38.596 10644 10644 D LemmyAPI: REQUEST: https://feddit.org/api/v3/post/list?type_=Local&sort=Active
02-13 08:22:38.596 10644 10644 D LemmyAPI: METHOD: HttpMethod(value=GET)
02-13 08:22:38.596 10644 10644 D LemmyAPI: COMMON HEADERS
02-13 08:22:38.596 10644 10644 D LemmyAPI: -> Accept: application/json
02-13 08:22:38.596 10644 10644 D LemmyAPI: -> Accept-Charset: UTF-8
02-13 08:22:38.596 10644 10644 D LemmyAPI: -> Authorization: ***
02-13 08:22:38.596 10644 10644 D LemmyAPI: -> User-Agent: Jerboa
02-13 08:22:38.596 10644 10644 D LemmyAPI: CONTENT HEADERS
02-13 08:22:38.596 10644 10644 D LemmyAPI: -> Content-Length: 0
02-13 08:22:38.596 10644 10644 D LemmyAPI: BODY Content-Type: null
02-13 08:22:38.596 10644 10644 D LemmyAPI: BODY START
02-13 08:22:38.596 10644 10644 D LemmyAPI:
02-13 08:22:38.596 10644 10644 D LemmyAPI: BODY END
02-13 08:22:38.596 10644 10644 D SiteViewModel: acc init for id: 928411
02-13 08:22:38.597 10644 10644 D LemmyAPI: REQUEST: https://feddit.org/api/v3/site
02-13 08:22:38.597 10644 10644 D LemmyAPI: METHOD: HttpMethod(value=GET)
02-13 08:22:38.597 10644 10644 D LemmyAPI: COMMON HEADERS
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Accept: application/json
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Accept-Charset: UTF-8
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Authorization: ***
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> User-Agent: Jerboa
02-13 08:22:38.597 10644 10644 D LemmyAPI: CONTENT HEADERS
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Content-Length: 0
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY Content-Type: null
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY START
02-13 08:22:38.597 10644 10644 D LemmyAPI:
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY END
02-13 08:22:38.597 10644 10644 D LemmyAPI: REQUEST: https://feddit.org/api/v3/user/unread_count
02-13 08:22:38.597 10644 10644 D LemmyAPI: METHOD: HttpMethod(value=GET)
02-13 08:22:38.597 10644 10644 D LemmyAPI: COMMON HEADERS
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Accept: application/json
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Accept-Charset: UTF-8
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Authorization: ***
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> User-Agent: Jerboa
02-13 08:22:38.597 10644 10644 D LemmyAPI: CONTENT HEADERS
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Content-Length: 0
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY Content-Type: null
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY START
02-13 08:22:38.597 10644 10644 D LemmyAPI:
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY END
02-13 08:22:38.597 10644 10644 D LemmyAPI: REQUEST: https://feddit.org/api/v3/user/report_count
02-13 08:22:38.597 10644 10644 D LemmyAPI: METHOD: HttpMethod(value=GET)
02-13 08:22:38.597 10644 10644 D LemmyAPI: COMMON HEADERS
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Accept: application/json
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Accept-Charset: UTF-8
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Authorization: ***
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> User-Agent: Jerboa
02-13 08:22:38.597 10644 10644 D LemmyAPI: CONTENT HEADERS
02-13 08:22:38.597 10644 10644 D LemmyAPI: -> Content-Length: 0
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY Content-Type: null
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY START
02-13 08:22:38.597 10644 10644 D LemmyAPI:
02-13 08:22:38.597 10644 10644 D LemmyAPI: BODY END
02-13 08:22:38.606 10644 10644 D jerboa  : got to inbox screen
02-13 08:22:38.656 10744 10744 W libc    : Access denied finding property "ro.debuggable"
--------- switch to events
02-13 08:22:38.655 10744 10744 I auditd  : avc=type=1400 audit(0.0:5369): avc:  denied  { read } for  comm="cat" name="u:object_r:userdebug_or_eng_prop:s0" dev="tmpfs" ino=779 scontext=u:r:untrusted_app:s0:c53,c257,c512,c768 tcontext=u:object_r:userdebug_or_eng_prop:s0 tclass=file permissive=0 bug=b/305600845 app=com.jerboa
--------- switch to main
02-13 08:22:38.655 10744 10744 W cat     : type=1400 audit(0.0:5369): avc:  denied  { read } for  name="u:object_r:userdebug_or_eng_prop:s0" dev="tmpfs" ino=779 scontext=u:r:untrusted_app:s0:c53,c257,c512,c768 tcontext=u:object_r:userdebug_or_eng_prop:s0 tclass=file permissive=0 bug=b/305600845 app=com.jerboa
--------- beginning of crash
02-13 08:22:38.678 10644 10676 E AndroidRuntime: FATAL EXCEPTION: arch_disk_io_1
02-13 08:22:38.678 10644 10676 E AndroidRuntime: Process: com.jerboa, PID: 10644
02-13 08:22:38.678 10644 10676 E AndroidRuntime: coil.network.HttpException: Exception in completion handler InvokeOnCompletion@9f08cd3[job@4867010] for StandaloneCoroutine{Completed}@4867010
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport.completeStateFinalization(Unknown Source:137)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport.finalizeFinishingState(Unknown Source:155)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport$ChildCompletion.invoke(Unknown Source:41)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport.completeStateFinalization(Unknown Source:44)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport.tryMakeCompleting(Unknown Source:53)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(Unknown Source:4)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.AbstractCoroutine.resumeWith(Unknown Source:13)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:31)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:112)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(Unknown Source:140)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlin.io.ByteStreamsKt.runBlocking(Unknown Source:104)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.appcompat.view.menu.CascadingMenuPopup$3$1.run(Unknown Source:33)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.room.TransactionExecutor$$ExternalSyntheticLambda0.run(Unknown Source:110)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1117)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Completed}@4867010, Dispatchers.Main.immediate]
02-13 08:22:38.678 10644 10676 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Method setCurrentState must be called on the main thread
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.enforceMainThreadIfNeeded(Unknown Source:42)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.setCurrentState(Unknown Source:7)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavBackStackEntry.updateState(Unknown Source:48)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavBackStackEntry.setMaxLifecycle(Unknown Source:7)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController.popEntryFromBackStack(Unknown Source:107)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController$executePopOperations$1.invoke(Unknown Source:22)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController$NavControllerNavigatorState.pop(Unknown Source:38)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController$NavControllerNavigatorState.popWithTransition(Unknown Source:189)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.compose.ComposeNavigator.popBackStack(Unknown Source:4)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController.executePopOperations(Unknown Source:61)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController.popBackStackInternal(Unknown Source:95)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController.navigate(SourceFile:47)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController.navigate$default(Unknown Source:84)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at androidx.navigation.NavController.navigate(SourceFile:139)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at com.jerboa.UtilsKt$$ExternalSyntheticLambda7.invoke(Unknown Source:81)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at com.jerboa.ui.components.home.BottomNavScreenKt$$ExternalSyntheticLambda3.invoke(Unknown Source:176)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at com.jerboa.ui.components.drawer.MainDrawerKt$$ExternalSyntheticLambda9.invoke(Unknown Source:28)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.InternalCompletionHandler$UserSupplied.invoke(Unknown Source:2)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.InvokeOnCompletion.invoke(Unknown Source:55)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	at kotlinx.coroutines.JobSupport.completeStateFinalization(Unknown Source:106)
02-13 08:22:38.678 10644 10676 E AndroidRuntime: 	... 15 more
--------- switch to events
02-13 08:22:38.690 10644 10644 I wm_on_top_resumed_lost_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=topStateChangedWhenResumed]
02-13 08:22:38.691 10644 10644 I wm_on_paused_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=performPause,time=0ms]
--------- switch to main
02-13 08:22:38.758 10644 10644 D VRI[MainActivity]: visibilityChanged oldVisibility=true newVisibility=false
02-13 08:22:38.768 10644 10718 D LemmyAPI: RESPONSE: 200
02-13 08:22:38.768 10644 10718 D LemmyAPI: METHOD: HttpMethod(value=GET)
02-13 08:22:38.768 10644 10718 D LemmyAPI: FROM: https://feddit.org/api/v3/user/unread_count
02-13 08:22:38.768 10644 10718 D LemmyAPI: COMMON HEADERS
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> access-control-allow-credentials: true
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> access-control-allow-headers: DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> access-control-allow-methods: GET, PUT, POST, DELETE, PATCH, OPTIONS
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> access-control-allow-origin: *
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> access-control-max-age: 1728000
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> cache-control: private
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> content-type: application/json
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> date: Thu, 13 Feb 2025 08:22:37 GMT
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> strict-transport-security: max-age=31536000; includeSubDomains
02-13 08:22:38.768 10644 10718 D LemmyAPI: -> vary: accept-encoding, Origin, Access-Control-Request-Method, Access-Control-Request-Headers
02-13 08:22:38.768 10644 10718 D LemmyAPI: BODY Content-Type: application/json
02-13 08:22:38.768 10644 10718 D LemmyAPI: BODY START
02-13 08:22:38.768 10644 10718 D LemmyAPI: {"replies":0,"mentions":0,"private_messages":0}
02-13 08:22:38.768 10644 10718 D LemmyAPI: BODY END
02-13 08:22:38.773 10644 10718 D LemmyAPI: RESPONSE: 200
02-13 08:22:38.773 10644 10718 D LemmyAPI: METHOD: HttpMethod(value=GET)
02-13 08:22:38.773 10644 10718 D LemmyAPI: FROM: https://feddit.org/api/v3/user/report_count
02-13 08:22:38.773 10644 10718 D LemmyAPI: COMMON HEADERS
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> access-control-allow-credentials: true
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> access-control-allow-headers: DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> access-control-allow-methods: GET, PUT, POST, DELETE, PATCH, OPTIONS
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> access-control-allow-origin: *
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> access-control-max-age: 1728000
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> cache-control: private
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> content-type: application/json
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> date: Thu, 13 Feb 2025 08:22:37 GMT
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> strict-transport-security: max-age=31536000; includeSubDomains
02-13 08:22:38.773 10644 10718 D LemmyAPI: -> vary: accept-encoding, Origin, Access-Control-Request-Method, Access-Control-Request-Headers
02-13 08:22:38.773 10644 10718 D LemmyAPI: BODY Content-Type: application/json
02-13 08:22:38.773 10644 10718 D LemmyAPI: BODY START
02-13 08:22:38.773 10644 10718 D LemmyAPI: {"comment_reports":0,"post_reports":0}
02-13 08:22:38.773 10644 10718 D LemmyAPI: BODY END
--------- switch to events
02-13 08:22:38.788 10644 10644 I viewroot_draw_event: [window=VRI[MainActivity],event=Not drawing due to not visible. Reason=!mAppVisible && !mForceDecorViewVisibility]
02-13 08:22:38.801 10644 10644 I wm_on_stop_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=LIFECYCLER_STOP_ACTIVITY,time=1ms]
02-13 08:22:38.852 10644 10644 I wm_on_destroy_called: [Token=130485253,Component Name=com.jerboa.MainActivity,Reason=performDestroy,time=1ms]
--------- switch to main
02-13 08:22:38.854 10644 10644 W WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=false callback=android.view.ViewRootImpl$$ExternalSyntheticLambda11@eb3647d
02-13 08:22:38.872 10644 10644 D LemmyAPI: REQUEST https://feddit.org/api/v3/post/list?type_=Local&sort=Active failed with exception: kotlinx.coroutines.JobCancellationException: Job was cancelled; job=SupervisorJobImpl{Cancelling}@86ecaaf
02-13 08:22:38.875 10644 10644 D LemmyAPI: REQUEST https://feddit.org/api/v3/site failed with exception: kotlinx.coroutines.JobCancellationException: Job was cancelled; job=SupervisorJobImpl{Cancelling}@b47cebc
02-13 08:22:40.205 10644 10676 I Process : Sending signal. PID: 10644 SIG: 9
```

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -16,7 +16,6 @@ plugins {
 apply(from = "update_instances.gradle.kts")
 
 android {
-    buildToolsVersion = "34.0.0"
     compileSdk = 34
 
     defaultConfig {
@@ -95,10 +94,6 @@ android {
     buildFeatures {
         compose = true
     }
-}
-
-composeCompiler {
-    featureFlags = setOf(ComposeFeatureFlag.StrongSkipping)
 }
 
 baselineProfile {

--- a/app/src/main/java/com/jerboa/feat/AccountVerificationState.kt
+++ b/app/src/main/java/com/jerboa/feat/AccountVerificationState.kt
@@ -467,7 +467,7 @@ fun Account.doIfReadyElseDisplayInfo(
     loginAsToast: Boolean = true,
     doAction: (Account) -> Unit,
 ) {
-    scope.launch {
+    scope.launch(Dispatchers.Main) {
         if (this@doIfReadyElseDisplayInfo.isReadyAndIfNotDisplayInfo(
                 appState,
                 ctx,

--- a/app/src/main/java/com/jerboa/ui/components/home/BottomNavScreen.kt
+++ b/app/src/main/java/com/jerboa/ui/components/home/BottomNavScreen.kt
@@ -61,6 +61,7 @@ import com.jerboa.ui.components.inbox.InboxScreen
 import com.jerboa.ui.components.person.PersonProfileScreen
 import com.jerboa.ui.components.registrationapplications.RegistrationApplicationsScreen
 import com.jerboa.ui.components.reports.ReportsScreen
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 
 enum class NavTab(
@@ -167,13 +168,15 @@ fun BottomNavScreen(
         selectedTab = tab
         val currentRoute = bottomNavController.currentDestination?.route
         if (currentRoute == tab.name && tab == NavTab.Home) {
-            scope.launch {
+            scope.launch(Dispatchers.Main) {
                 homeViewModel.lazyListState.animateScrollToItem(0)
             }
         } else {
-            bottomNavController.navigate(tab.name) {
-                launchSingleTop = true
-                popUpTo(bottomNavController.graph.id) // To make back button close the app.
+            scope.launch(Dispatchers.Main) {
+                bottomNavController.navigate(tab.name) {
+                    launchSingleTop = true
+                    popUpTo(bottomNavController.graph.id) // To make back button close the app.
+                }
             }
         }
     }

